### PR TITLE
[web3]: fix eth.getBalance return type

### PR DESCRIPTION
**Pull request by dkent600**
fixes eth.getBalance return type (`number` => `BigNumber`)

docs: https://web3js.readthedocs.io/en/1.0/web3-eth.html?#getbalance

But note that the docs that specify `string` are in conflict with the code that specifies `BigNumber`:

```
        new Method({
            name: 'getBalance',
            call: 'eth_getBalance',
            params: 2,
            inputFormatter: [formatter.inputAddressFormatter, formatter.inputDefaultBlockNumberFormatter],
            outputFormatter: formatter.outputBigNumberFormatter
        }),
```

I would supply a link to the repository for the above code, but haven't been able to find the repository.

### CODE REVIEW

**Review by phra — CHANGES_REQUESTED** [review]
add unit tests to it.

**Review by levino — CHANGES_REQUESTED** [review]
Please create an overload for the `getBalance` function: One that has a callback as a required argument, but returns `void`. And an overload that does not accept a callback but always returns a promise.

And as phra said, please add two simple unit tests for this.

**Comment by dkent600** [comment]
@Levino @phra 

Suggested changes are done, thanks.

**Review by levino — APPROVED** [review]
Perfect, thank you

File: types/web3/eth/index.d.ts [hunk 1]
```diff
@@ -85,11 +85,15 @@ export default interface Eth {
...
+    getBalance(
+        address: string,
+        defaultBlock?: BlockType
+    ): Promise<BigNumber>;
     getBalance(
         address: string,
         defaultBlock?: BlockType,
-        cb?: Callback<number>
-    ): Promise<number>;
+        cb?: Callback<BigNumber>
```
Comment (zlumer) [inline types/web3/eth/index.d.ts]: If `cb` parameter is left optional, it creates two conflicting method signatures:
`getBalance(address: string, defaultBlock?: BlockType): Promise<BigNumber>`
and
`getBalance(address: string, defaultBlock?: BlockType): void`

`cb` should be a required argument to make intent clear.

**Review by zlumer — CHANGES_REQUESTED** [review]

File: types/web3/eth/index.d.ts (hunk 1 above)
Comment (Levino) [reply types/web3/eth/index.d.ts]: Good catch

**Review by phra — APPROVED** [review]

**Review by zlumer — APPROVED** [review]

**Pull request merged**